src/app/lists/dataview.tsx

```diff
@@ -0,0 +1,202 @@
+"use client";
+
+import Table, { Datum } from "@/components/Tables/Table";
+import { Database } from "@/database_types_file";
+import React, { useState } from "react";
+import { styled } from "styled-components";
+import EditModal from "@/app/lists/edit_modal";
+import supabase from "@/supabase";
+import ConfirmDialog from "@/components/Modal/Confirm";
+import Snackbar from "@mui/material/Snackbar/Snackbar";
+import Alert from "@mui/material/Alert/Alert";
+
+export type ListRow = Database["public"]["Tables"]["lists"]["Row"];
+
+const TableDiv = styled.div`
+    margin: 20px;
+    padding: 20px;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+    border: solid 1px ${(props) => props.theme.surfaceForegroundColor};
+    border-radius: 1rem;
+`;
+
+const StyledTable = styled(Table)`
+    width: 100%;
+    background-color: transparent;
+`;
+
+const StyledAddButton = styled.button`
+    margin: 10px 5px 5px 0;
+    height: 2rem;
+    width: 5rem;
+`;
+
+type Props = {
+    data: ListRow[] | null;
+};
+
+export const headers: [string, string][] = [
+    ["item_name", "Description"],
+    ["1_quantity", "Single"],
+    ["2_quantity", "Family of 2"],
+    ["3_quantity", "Family of 3"],
+    ["4_quantity", "Family of 4"],
+    ["5_quantity", "Family of 5"],
+    ["6_quantity", "Family of 6"],
+    ["7_quantity", "Family of 7"],
+    ["8_quantity", "Family of 8"],
+    ["9_quantity", "Family of 9"],
+    ["10_quantity", "Family of 10+"],
+];
+
+export const tooltips: [string, string][] = [
+    ["1_notes", "Single"],
+    ["2_notes", "Family of 2"],
+    ["3_notes", "Family of 3"],
+    ["4_notes", "Family of 4"],
+    ["5_notes", "Family of 5"],
+    ["6_notes", "Family of 6"],
+    ["7_notes", "Family of 7"],
+    ["8_notes", "Family of 8"],
+    ["9_notes", "Family of 9"],
+    ["10_notes", "Family of 10+"],
+];
+
+const ListsDataView: React.FC<Props> = ({ data: rawData }) => {
+    // null => add, undefined => modal closed
+    const [modal, setModal] = useState<Datum | null | undefined>(undefined);
+    const [toDelete, setToDelete] = useState<number | null>(null);
+    // need another setState otherwise the modal content changes before the close animation finishes
+    const [toDeleteModalOpen, setToDeleteModalOpen] = useState<boolean>(false);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    if (rawData === null) {
+        throw new Error("No data found");
+    }
+
+    const remapTooltips = (
+        row: ListRow
+    ): Datum & { unmappedTooltips: { [key: string]: string | null } } => {
+        const data: { [key: string]: string | null } = {
```

Comment:
{ [key: string]: string | null }, is being used across both files multiple times, Might be better to put this into a type?